lib/ecto/repo.ex

```diff
@@ -567,7 +567,9 @@ defmodule Ecto.Repo do
       none is specified, the conflict target is left up to the database
       and is usually made of primary keys and/or unique/exclusion constraints.
       May also be `{:constraint, constraint_name_as_atom}` in databases
-      that support the "ON CONSTRAINT" expression.
+      that support the "ON CONSTRAINT" expression or `{:fragment, binary_fragment}`
+      for targets that require expressions on columns rather than raw column names,
+      e.g. ON CONFLICT (coalesce(u.firstname, ""), coalesce(u.lastname, ""))
```

Comment:
Maybe use backticks here `ON CONFLICT (coalesce(u.firstname, ""), coalesce(u.lastname, ""))`?

I would also include a dot at the end of the sentence :)